crates/extensions/c8y_mapper_ext/src/operations/handlers/device_profile.rs

```diff
@@ -0,0 +1,1208 @@
+use anyhow::Context;
+use c8y_api::smartrest;
+use c8y_api::smartrest::smartrest_serializer::CumulocitySupportedOperations;
+use tedge_api::device_profile::DeviceProfileCmd;
+use tedge_api::CommandStatus;
+use tedge_mqtt_ext::MqttMessage;
+use tracing::warn;
+
+use super::EntityTarget;
+use super::OperationContext;
+use super::OperationError;
+use super::OperationOutcome;
+
+impl OperationContext {
+    pub async fn handle_device_profile_state_change(
+        &self,
+        target: &EntityTarget,
+        cmd_id: &str,
+        message: &MqttMessage,
+    ) -> Result<OperationOutcome, OperationError> {
+        if !self.capabilities.device_profile {
+            warn!("Received a device_profile command, however, device_profile feature is disabled");
+            return Ok(OperationOutcome::Ignored);
+        }
+
+        let command = match DeviceProfileCmd::try_from_bytes(
+            target.topic_id.to_owned(),
+            cmd_id.into(),
+            message.payload_bytes(),
+        )
+        .context("Could not parse command as a device profile command")?
+        {
+            Some(command) => command,
+            None => {
+                // The command has been fully processed
+                return Ok(OperationOutcome::Ignored);
+            }
+        };
+
+        let sm_topic = &target.smartrest_publish_topic;
+
+        match command.status() {
+            CommandStatus::Executing => Ok(OperationOutcome::Executing),
```

Comment:
Also for `Executing`, you need to send a SmartREST message `121,false`. Refer to the [doc](https://cumulocity.com/docs/device-integration/fragment-library/#device-profile)

> ...
> 2. Set operation status to EXECUTING
> 501,c8y_DeviceProfile
> 3. Set target profile
> 121,false,
> ...

Reply:
Also, since it's a SmartREST message that updates `inventory`, I recommend you to add a function to the `crates/core/c8y_api/src/smartrest/inventory.rs` file and call the function for `Executing` and `Succssful`.